enhancements/windows-containers/monitoring-windows-nodes.md

```diff
@@ -0,0 +1,182 @@
+ enhancement](https://github.com/openshift/enhancements/blob/master/enhancements/windows-containers/windows-machine-config-operator.md#justification)
+for more details.
+
+### Risks and Mitigations
+
+* Running `windows_exporter` as a Windows Service, posses a risk of having
+  inadequate resources to run the service if the Windows node is overwhelmed
+  with workload containers. This can be mitigated by leveraging [priority
+   classes](https://docs.microsoft.com/en-us/windows/win32/procthread/scheduling-priorities) for
+  Windows processes. This is similar to what is being done for other [Windows
+  services](https://issues.redhat.com/browse/WINC-534).
+  
+* One of the risks with the current approach is renaming Windows metrics to
+  display pod graphs. The pod metrics for Linux come from cAdvisor. However, we
+  do not get same metrics from cAdvisor for Windows nodes. This becomes a
+  hindrance to display pod graphs by creating custom recording rules to use same
+  console queries as Linux workloads. To mitigate this, use metrics exposed by
+  the windows_exporter to display pod graphs as mentioned in the [Future
+   Plans](https://github.com/openshift/enhancements/blob/master/enhancements/windows-containers/monitoring-windows-nodes.md#future-plans)
+  is required. This also requires changes in console queries that support OS
+  specific metrics.
+
+## Design Details
+
+As we are not able to run windows_exporter as a [container](https://github.com/openshift/enhancements/blob/master/enhancements/windows-containers/monitoring-windows-node.md#justification)
+on the Windows Node, to capture data from windows_exporter, WMCO creates a
+`windows-machine-config-operator-metrics` Service without selectors and
+manually defines Endpoints object for that service. The Endpoints object has
+entries for the endpoints `<internal-ip>:9182/metrics`, exposed by
+windows_exporter for every Windows node. Once the Service and Endpoints
+object is created, WMCO ensures that a Service Monitor for `windows-machine-config-operator-metrics`
+Service is running so that the Prometheus operator can discover the targets
+created above to scrape Windows metrics. Following design details reflect the
+current approach and future plans to enable monitoring support for Windows.
+
+### Current State
+
+To enable basic monitoring support for Windows node, WMCO has done the
+following:
+
+* Build and add windows_exporter binary to WMCO payload.
+* Install windows_exporter on the Windows nodes and ensuring
+  that it runs as a Windows service.
+* Add `openshift.io/cluster-monitoring=true` label to the
+ `openshift-windows-machine-config-operator` namespace so that cluster
+  monitoring stack will pick up the Service Monitor created by WMCO.
+* Add privileges to WMCO to create Services, Endpoints, Service Monitor in
+  the `openshift-windows-machine-config-operator` namespace.
+* Create a Service and Endpoints object in `openshift-windows-machine-config
```

Comment:
In general, the metrics should be exposed over HTTPS and the endpoint should authenticate the request (either using RBAC or mutual TLS). When running inside a pod, we can leverage the CA signer for providing TLS cert/key to the service + [kube-rbac-proxy](https://github.com/brancz/kube-rbac-proxy) for authentication (and possibly authorization).
In the case of Windows, I assume that it's going to be more challenging since the service isn't managed with the Kubernetes API. If it isn't achievable, it should at least be stated in the proposal IMO.

Reply:
I agree with @simonpasquier  that it needs to be called out. `windows_exporter` recently added support for [HTTPS configuration](https://github.com/prometheus-community/windows_exporter/commit/cdc81b03d5654f115aa55501b7aaa1b99991dc48). @VaishnaviHire please mention that in the future plans and add a spike to investigate adding [web config](https://github.com/prometheus/exporter-toolkit/blob/master/docs/web-configuration.md) for TLS,

Reply:
Updated to reflect this `future plans`